application/controllers/upload.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+session_start();
```

Comment:
If someone try to access this page through `url` then whatever logic will perform but at first `session` will be start. So please check it and if you want to share something then please share.

Reply:
Agreed. I will start the session inside the function after the validation is succesful